x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/elasticsearch_monitor_states_adapter.ts

```diff
@@ -216,11 +217,12 @@ export class ElasticsearchMonitorStatesAdapter implements UMMonitorStatesAdapter
       );
       const { result, statusFilter } = queryRes;
       searchAfter = queryRes.searchAfter;
-      monitors.push(...this.getMonitorBuckets(result, statusFilter));
-      if (get<number>(result, 'hits.total.value', 0) === 0) {
-        break;
+      const monitorBucketsToAdd = this.getMonitorBuckets(result, statusFilter);
+      monitors.push(...monitorBucketsToAdd);
+      if (monitors.length > STATES.LEGACY_STATES_QUERY_SIZE) {
+        monitors = monitors.slice(0, STATES.LEGACY_STATES_QUERY_SIZE);
       }
-    } while (searchAfter !== null && monitors.length < STATES.LEGACY_STATES_QUERY_SIZE);
+    } while (!!searchAfter && monitors.length < STATES.LEGACY_STATES_QUERY_SIZE);
```

Comment:
Do we actually need the double negation here? Shouldn't a plain `searchAfter` be the boolean equivalent?

Reply:
It should be, yes, I'm just in the habit of explicitly casting to a boolean. We don't need it. It's probably a leftover result of hacking more than anything else.